swarm/api/api.go

```diff
@@ -308,7 +308,7 @@ func (self *Api) Put(content, contentType string, toEncrypt bool) (k storage.Key
 // Get uses iterative manifest retrieval and prefix matching
 // to resolve basePath to content using dpa retrieve
 // it returns a section reader, mimeType, status and an error
-func (self *Api) Get(key storage.Key, path string) (reader storage.LazySectionReader, mimeType string, status int, err error) {
+func (self *Api) Get(key storage.Key, path string) (reader *storage.LazyChunkReader, mimeType string, status int, err error) {
```

Comment:
Maybe I miss something, but I don't get why you changed from `LazySectionReader` interface to `LazyChunkReader` struct here and in the other places you modified.

Reply:
http://idiomaticgo.com/post/best-practice/accept-interfaces-return-structs/ - TLDR - easier to read the code.

Reply:
@janos yes, we are no longer using the Joiner interface and it has been removed. If we need it, then we fix this code to return an interface.